skimage/filters/_multiotsu.pyx

```diff
@@ -0,0 +1,54 @@
+#cython: cdivision=True
+#cython: boundscheck=False
+#cython: nonecheck=False
+#cython: wraparound=False
+import numpy as np
+
+cimport numpy as cnp
+cimport cython
+from itertools import combinations
+
+def _find_threshold_multiotsu(double [:, ::1] var_btwcls,
+                              Py_ssize_t classes,
+                              Py_ssize_t bins,
+                              double [::1] aux_thresh):
+    """
+    Cython utility function for finding thresholds in multi-Otsu algorithm.
+    This function is only called by filters.threshold_multiotsu.
+
+    Parameters
+    ----------
+    var_btwcls : 2-d array
+        array of variance between classes
+    classes : int
+        number of classes to be segmented
+    bins : int
+        number of bins used in the histogram
+    aux_thresh : array
+        thresholds to be returned
+    """
+    cdef Py_ssize_t idd
+    cdef Py_ssize_t sh0, sh1
+    cdef double part_sigma = 0
+    # max_sigma is the maximum variance between classes.
+    cdef double max_sigma = 0
+    cdef Py_ssize_t [:, ::1] _tmp = np.array(list(
+                                combinations(range(1, bins - 2), classes - 1)))
+    sh0 = _tmp.shape[0]
+    sh1 = _tmp.shape[1]
+    cdef Py_ssize_t [:, :] idx_tuples = np.zeros((sh0, sh1 + 2), 
+                                                  dtype=np.int)
+    idx_tuples[:, 1:sh1 + 1] = _tmp[:]
+    idx_tuples[:, sh1 + 1] = bins - 1
+    cdef Py_ssize_t [::1] idx_tuple
```

Comment:
You can add your allocation for 
```
    py_aux_thresh = np.zeros(classes-1, dtype=np.intp)
    cdef Py_ssize_t [::1] aux_thresh = py_aux_thresh
```
here

Reply:
We should not be mixing types here.  It is likely that `Py_ssize_t` happens to be the right type, but you can write this as:

```
cdef cnp.int64_t [::1] idx_tuple = np.zeros(..., dtype=np.int64)
```

Then the match is exactly right.

Reply:
Thanks @stefanv, This is one of my big pet peves with data types. Everybody likes to rename them.....

Reply:
:crown: I crown you Guardian of the Types.  (@jni is the Guardian of the Dimensions, so it's good to we can separate our pet peeves so well).